gflownet/gflownet.py

```diff
@@ -311,178 +369,145 @@ def step(
         actions : list
             A list of actions to be executed on each env of envs.
 
-        is_forward : bool
-            True if sampling is forward. False if backward.
-
-        temperature : float
-            Temperature to adjust the logits by logits /= temperature
+        backward : bool
+            True if sampling is backward. False (forward) by default.
         """
         assert len(envs) == len(actions)
         if not isinstance(envs, list):
             envs = [envs]
-        if is_forward:
+        if backward:
             _, actions, valids = zip(
-                *[env.step(action) for env, action in zip(envs, actions)]
+                *[env.step_backwards(action) for env, action in zip(envs, actions)]
             )
         else:
-            valids = []
-            for env, action in zip(envs, actions):
-                parents, parents_a = env.get_parents(action=action)
-                if action in parents_a:
-                    state_next = parents[parents_a.index(action)]
-                    env.set_state(state_next, done=False)
-                    env.n_actions -= 1
-                    valids.append(True)
-                else:
-                    valids.append(False)
+            _, actions, valids = zip(
+                *[env.step(action) for env, action in zip(envs, actions)]
+            )
         return envs, actions, valids
 
     # @profile
```

Comment:
Can this be removed?

Reply:
Not sure I understand. Do you mean whether `valids.append(False)` can be removed? (Yes, it can because `valids` is filled now by the `zip` loop. Or are you suggesting to remove this whole thing?

Reply:
Oh, sorry, I was referring specifically to `# @profile` below - not what you've changed in this PR (I am not sure if that's an artifact someone forgot to remove earlier or not).